### ISSUE TITLE
Add native support for the Blackhole fused router

### ISSUE THREAD

**Issue by ifabijanic-tt** [tenstorrent/tt-metal#57186]
### Problem

We should implement and benchmark native Blackhole support for
`ttnn.experimental.topk_router_gpt`, the fused GPT-OSS router that combines
matrix multiplication, bias, top-k selection, and softmax.

This is a follow-up to #56949 and the architecture-guard fix also proposed in
#56956. The guard prevents the crash by routing Blackhole through the existing
`linear` + `topk` + `softmax` implementation. Native fused support would allow
Blackhole to benefit from reduced operation overhead and intermediate tensor
traffic. The performance benefit has not yet been measured.

The current kernel requires **4 groups × 3 DRAM-aligned cores = 12 cores**.
Blackhole provides 8 such assignments, so direct execution fails with:

```text
topk_router_gpt requires at least 12 DRAM-aligned cores, got 8
```

Each group handles 32 of the 128 experts. Two sender cores compute partial
matrix products; a worker computes its own partial product and combines all
three. One worker also collects the four groups' logits and performs top-k and
softmax. Core roles, partial-result buffers, and semaphore waits encode this
three-core layout, so changing the core-count assertion alone is insufficient.

### One proposed solution

Start by evaluating **4 groups × 2 cores** on Blackhole: one sender and one worker
per group, retaining the four-group collector and the existing 128-expert output
contract. Keep the existing 4×3 configuration on Wormhole.

1. Parameterize the group size, sender count, worker/collector placement, and
   hidden-dimension partitioning in the program factory. Handle uneven tile
   partitions and reject unsupported configurations during operation validation.
2. Update partial-result accumulation, circular-buffer sizing, and semaphore
   counts consistently. Preserve the matching buffer layouts required by the
   existing cross-core writes.
3. Preserve output shapes, layouts, dtypes, expert-index ordering, and routing
   weight semantics. Verify repeated execution, program-cache reuse with new
   tensor addresses, and trace capture/replay.
4. Update both model router implementations to permit Blackhole fusion for
   validated configurations, retaining the generic fallback for other cases.
   Enable fusion by default only after correctness and performance validation.

IMPORTANT NOTE: The 4×2 design is a starting proposal, not a necessarily validated implementation!

### One Alternative

- Evaluate a 12-worker layout that selects additional compute cores outside the
  current DRAM-aligned assignment. This needs separate placement and memory
  traffic analysis; eight DRAM-aligned assignments do not limit the chip to
  eight compute cores.
